serverPostgreSQL.js

```diff
@@ -0,0 +1,297 @@
+  .post("/postCohort", postCohort)
+  .delete("/deleteLOs", deleteLOs)
+  .start({ port: PORT });
+console.log(`Server running on http://localhost:${PORT}`);
+
+function allowCors() {
+  return abcCors({
+    origin: `http://localhost:3000`,
+    headers: allowedHeaders,
+    credentials: true,
+  });
+}
+
+async function getLOs(server) {
+  const { user_id } = await server.params;
+  const query = `
+    SELECT *
+    FROM results
+    WHERE user_id = $1
+  `;
+  const LOs = (await client.queryObject({ text: query, args: [user_id] })).rows;
+  if (LOs.length !== 0) {
+    return server.json(LOs, 200);
+  } else {
+    return server.json({ error: "Student does not exist" }, 400);
+  }
+}
+
+async function getStudents(server) {
+  const { cohort_id } = await server.params;
+  const query = `
+    SELECT DISTINCT email, user_id
+    FROM results
+    WHERE cohort_id = $1
+  `;
+  const LOs = (await client.queryObject({ text: query, args: [cohort_id] }))
+    .rows;
+
+  return server.json(LOs, 200);
+}
+
+async function getStudentData(server) {
+  const { user_id } = await server.params;
+  const query = `
+  SELECT *
+  FROM results
+  WHERE user_id = $1
+  ORDER BY topic ASC
+  `;
+  const LOs = (await client.queryObject({ text: query, args: [user_id] })).rows;
+  return server.json(LOs, 200);
+}
+
+async function getCohortLOs(server) {
+  const { cohort_id } = await server.params;
+  const query = `
+    SELECT *
+    FROM learning_objectives
+    WHERE cohort_id = $1
+    ORDER BY topic ASC
+  `;
+  const cohortLOs = (
+    await client.queryObject({ text: query, args: [cohort_id] })
+  ).rows;
+  return server.json(cohortLOs);
+}
+
+async function getCohorts(server) {
+  const query = `
+    SELECT DISTINCT cohort_id 
+    FROM learning_objectives
+    ORDER BY cohort_id ASC
+  `;
+  const cohorts = (await client.queryObject(query)).rows;
+  return server.json(cohorts, 200);
+}
+
+async function getTopicsOnly(server) {
+  const { user_id } = await server.params;
+  const query = `
+    SELECT DISTINCT topic
+    FROM results
+    WHERE user_id = $1
+  `;
+  const cohortTopics = (
+    await client.queryObject({ text: query, args: [user_id] })
+  ).rows;
+  if (cohortTopics) {
+    return server.json(cohortTopics, 200);
+  } else {
+    return server.json({ error: "Topic list does not exist." }, 400);
+  }
+}
+
+async function postLO(server) {
+  const { cohort_id, topic, learning_objective } = await server.body;
+  const query = `
+    INSERT INTO learning_objectives(cohort_id, topic, learning_objective)
+    VALUES ($1, $2, $3)
+  `;
+
+  await client.queryObject({
+    text: query,
+    args: [cohort_id, topic, learning_objective],
+  }).rows;
+
+  const check = (
+    await client.queryObject({
+      text: `
+        SELECT DISTINCT(users.email), users.cohort_id, users.id 
+        FROM learning_objectives JOIN users ON users.cohort_id = learning_objectives.cohort_id 
+        WHERE users.cohort_id = $1
```

Comment:
Good work changing the formatting for postgreSQL

Reply:
yes i agree